jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/json/JsonCfnParser.kt

```diff
@@ -0,0 +1,395 @@
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnConditionsNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnExpressionNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnFirstLevelMappingNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnFunctionNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnGlobalsNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnIntrinsicFunction
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnMappingValue
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnMappingsNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnMetadataNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnNameValueNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnNamedNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnObjectValueNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnOutputNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnOutputsNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnParameterNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnParametersNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnParsedFile
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnProblem
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourceConditionNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourceDependsOnNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourceNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourcePropertiesNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourcePropertyNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourceTypeNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnResourcesNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnRootNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnScalarValueNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnSecondLevelMappingNode
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnSection
+import software.aws.toolkits.jetbrains.services.cloudformation.CfnTransformNode
+
+class JsonCfnParser private constructor() {
+    private val problems = mutableListOf<CfnProblem>()
+    private val nodeMap = BidirectionalMap<CfnNode, PsiElement>()
+
+    fun handleRoot(root: JsonObject): CfnRootNode {
+        val translatedSections = root.propertyList.mapNotNull { translateSection(it) }
+
+        return CfnRootNode(
+            translatedSections.findSection<CfnMetadataNode>(),
```

Comment:
Couldn't cfnrootnote have a function like "buildFromNodes" that take a list of cfnnodes and do this internally? otherwise this has to be duplicated in the yaml one

Reply:
I am going to to thing long on avoiding copy paste in the YAML code.

Original code has some divergence due to it assumed SAM was YAML only. We are removing that so it is even more similar now